src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -440,37 +468,68 @@ def initialize_resource(resc):
     return ret
 
 
-def ignore_job(jobid):
+def printjob_info(jobid, include_attributes=False):
     """
-    Helper function to ignore suspended jobs when removing
-    currently assigned resources
+    Use printjob to acquire the job information
     """
-    printjob_cmd = os.path.join(PBS_EXEC, 'bin', 'printjob')
-    job_file = os.path.join(PBS_MOM_HOME, 'mom_priv', 'jobs', '%s.JB' % jobid)
-    if not os.path.isfile(job_file):
-        pbs.logmsg(pbs.EVENT_DEBUG4, "File not found: %s" % (job_file))
-        return False
-    cmd = [printjob_cmd, job_file]
-    substate = None
-    # Get job substate based on printjob output
+    info = {}
+    jobfile = os.path.join(PBS_MOM_JOBS, '%s.JB' % jobid)
+    if not os.path.isfile(jobfile):
+        pbs.logmsg(pbs.EVENT_DEBUG4, "File not found: %s" % (jobfile))
+        return info
+    cmd = [os.path.join(PBS_EXEC, 'bin', 'printjob')]
+    if not include_attributes:
+        cmd.append('-a')
+    cmd.append(jobfile)
     try:
-        pbs.logmsg(pbs.EVENT_DEBUG4, "cmd: %s" % cmd)
-        # Collect the job substate information
+        pbs.logmsg(pbs.EVENT_DEBUG4, "Running: %s" % cmd)
         process = subprocess.Popen(cmd, shell=False,
                                    stdout=subprocess.PIPE,
                                    stderr=subprocess.PIPE)
         out, err = process.communicate()
-        # Find the job substate
-        regex = re.compile(r"substate:\s+(0x[0-9a-f]+)")
-        substate = regex.search(out)
-    except Exception as exc:
-        pbs.logmsg(pbs.EVENT_DEBUG, "%s: Unexpected error: %s" %
-                   (caller_name(), exc))
-    if substate:
-        substate = substate.group(1)
-        pbs.logmsg(pbs.EVENT_DEBUG4,
-                   "Job %s has substate %s" % (jobid, substate))
-        return substate in ['0x2b', '0x2d', 'unknown']
+    except KeyboardInterrupt:
+        raise
+    except Exception, exc:
```

Comment:
That won't work with some older versions of PBS Pro. There are still users with Python 2.5 that want to make use of the cgroup hook. We'll have to use the "as" syntax when we migrate to Python 3.

Reply:
I'm going to have to defer to @scc138 for a decision. There was an issue related to this recently.

Reply:
Maybe @subhasisb should weigh in on the decision as well?

Reply:
While it would be convenient if this hook would work with PBS Pro 13.x where Python 2.5 is present it is not a requirement.  Doing the right thing for the targeted Python versions (2.6 and up, less than 3.0) and being future proof are more important.

Reply:
I'll update all of the "Exception, exc:" to "Exception as exc:" in the next push.